### PR TITLE
Graduate WinOverlay to Beta

### PR DESCRIPTION
**Pull request by ksubrmnn**
**What type of PR is this?**

/kind feature
/kind api-change

**What this PR does / why we need it**:

This PR updates the WinOverlay flag to beta. Overlay networking in Windows has had stable test passes for over a year. This feature has been integrated with multiple CNIs such as Calico, OVN, and Flannel.

Test passes:

https://testgrid.k8s.io/sig-windows-networking#containerd-overlay-windows-master
https://testgrid.k8s.io/sig-windows-networking#flannel-overlay-windows-master

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
WinOverlay feature graduated to beta
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by ksubrmnn** [comment]
cc @marosset @daschott @jsturtevant @michmike

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -620,7 +620,7 @@ const (
 	SetHostnameAsFQDN featuregate.Feature = "SetHostnameAsFQDN"
 
 	// owner: @ksubrmnn
-	// alpha: v1.14
```
Comment (marosset) [inline pkg/features/kube_features.go:623]: Can you leave in //alpha: v1.14 like some of the more recent updates above to maintain some history here?

**Review by michmike — APPROVED** [review]
/lgtm
/approve

**Pull request merged**